example-app/SantokuApp/src/screens/demo/cache/FileInfo.tsx

```diff
@@ -0,0 +1,87 @@
+import * as FileSystem from 'expo-file-system';
+import React, {useCallback, useEffect, useState} from 'react';
+import {StyleSheet, View} from 'react-native';
+import {Icon, Text} from 'react-native-elements';
+
+import {useCacheDirectory} from './useCacheDirectory';
+
+type FileInfoProps = {
+  fileInfo: FileSystem.FileInfo;
+  currentDepth: number;
+};
+
+const maxDepth = 10;
+
+const FileInfo: React.FC<FileInfoProps> = props => {
+  const {fileInfo, currentDepth} = props;
+  const {readDirectoryItemsFileInfoAsync} = useCacheDirectory();
+  const [childFileInfos, setChildFileInfos] = useState<FileSystem.FileInfo[]>([]);
+
+  const getFileNameFromUri = useCallback((uri: string) => {
+    return uri.endsWith('/') ? uri.split('/').slice(-2)[0] : uri.split('/').slice(-1)[0];
+  }, []);
```

Comment:
`getFileNameFromUri `関数は内部ステートを参照しておらず、フック定義する必要もないので、コンポーネント関数外で定義して使用するのは如何でしょうか。

```suggestion
const getFileNameFromUri = (uri: string) => {
  return uri.endsWith('/') ? uri.split('/').slice(-2)[0] : uri.split('/').slice(-1)[0];
};

const FileInfo: React.FC<FileInfoProps> = props => {
  const {fileInfo, currentDepth} = props;
  const {readDirectoryItemsFileInfoAsync} = useCacheDirectory();
  const [childFileInfos, setChildFileInfos] = useState<FileSystem.FileInfo[]>([]);
```